libs/ml/benchmark/ops/ops.cpp

```diff
@@ -17,156 +17,3605 @@
+#include "ml/ops/mask_fill.hpp"
 #include "ml/ops/matrix_multiply.hpp"
+#include "ml/ops/max_pool_1d.hpp"
+#include "ml/ops/max_pool_2d.hpp"
+#include "ml/ops/maximum.hpp"
 #include "ml/ops/multiply.hpp"
+#include "ml/ops/one_hot.hpp"
 #include "ml/ops/ops.hpp"
+#include "ml/ops/prelu_op.hpp"
+#include "ml/ops/reduce_mean.hpp"
+#include "ml/ops/reshape.hpp"
+#include "ml/ops/slice.hpp"
 #include "ml/ops/sqrt.hpp"
 #include "ml/ops/squeeze.hpp"
 #include "ml/ops/subtract.hpp"
+#include "ml/ops/switch.hpp"
+#include "ml/ops/tanh.hpp"
+#include "ml/ops/top_k.hpp"
+#include "ml/ops/transpose.hpp"
 #include "vectorise/fixed_point/fixed_point.hpp"
 
 #include "benchmark/benchmark.h"
 
 #include <vector>
 
+template <class T, int N>
+void BM_AbsForward(benchmark::State &state)
+{
+  using TensorType    = typename fetch::math::Tensor<T>;
+  using VecTensorType = typename fetch::ml::ops::Ops<TensorType>::VecTensorType;
+
+  fetch::math::Tensor<T> input({1, N});
+  fetch::math::Tensor<T> output({1, N});
+
+  // Fill tensors with random values
+  input.FillUniformRandom();
+  output.FillUniformRandom();
+
+  VecTensorType inputs;
+  inputs.emplace_back(std::make_shared<TensorType>(input));
+  fetch::ml::ops::Abs<fetch::math::Tensor<T>> abs;
+
+  for (auto _ : state)
+  {
+    abs.Forward(inputs, output);
+  }
+}
+
+BENCHMARK_TEMPLATE(BM_AbsForward, float, 2)->Unit(benchmark::kNanosecond);
+BENCHMARK_TEMPLATE(BM_AbsForward, float, 256)->Unit(benchmark::kMicrosecond);
+BENCHMARK_TEMPLATE(BM_AbsForward, float, 512)->Unit(benchmark::kMicrosecond);
+BENCHMARK_TEMPLATE(BM_AbsForward, float, 1024)->Unit(benchmark::kMicrosecond);
+BENCHMARK_TEMPLATE(BM_AbsForward, float, 2048)->Unit(benchmark::kMicrosecond);
+BENCHMARK_TEMPLATE(BM_AbsForward, float, 4096)->Unit(benchmark::kMicrosecond);
+
+BENCHMARK_TEMPLATE(BM_AbsForward, double, 2)->Unit(benchmark::kNanosecond);
+BENCHMARK_TEMPLATE(BM_AbsForward, double, 256)->Unit(benchmark::kMicrosecond);
+BENCHMARK_TEMPLATE(BM_AbsForward, double, 512)->Unit(benchmark::kMicrosecond);
+BENCHMARK_TEMPLATE(BM_AbsForward, double, 1024)->Unit(benchmark::kMicrosecond);
+BENCHMARK_TEMPLATE(BM_AbsForward, double, 2048)->Unit(benchmark::kMicrosecond);
+BENCHMARK_TEMPLATE(BM_AbsForward, double, 4096)->Unit(benchmark::kMicrosecond);
+
+BENCHMARK_TEMPLATE(BM_AbsForward, fetch::fixed_point::fp32_t, 2)->Unit(benchmark::kNanosecond);
+BENCHMARK_TEMPLATE(BM_AbsForward, fetch::fixed_point::fp32_t, 256)->Unit(benchmark::kMicrosecond);
+BENCHMARK_TEMPLATE(BM_AbsForward, fetch::fixed_point::fp32_t, 512)->Unit(benchmark::kMicrosecond);
+BENCHMARK_TEMPLATE(BM_AbsForward, fetch::fixed_point::fp32_t, 1024)->Unit(benchmark::kMicrosecond);
+BENCHMARK_TEMPLATE(BM_AbsForward, fetch::fixed_point::fp32_t, 2048)->Unit(benchmark::kMicrosecond);
+BENCHMARK_TEMPLATE(BM_AbsForward, fetch::fixed_point::fp32_t, 4096)->Unit(benchmark::kMicrosecond);
+
+BENCHMARK_TEMPLATE(BM_AbsForward, fetch::fixed_point::fp64_t, 2)->Unit(benchmark::kNanosecond);
```

Comment:
it doesn't look like you have any fp128_t benchmarks?

Reply:
oops. missed that. will add them